File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/fragments/BridgeTextFragment.java
package com.facebook.react.views.text.fragments;

import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.uimanager.ReactStylesDiffMap;
import com.facebook.react.uimanager.ViewProps;
import com.facebook.react.views.text.TextAttributeProps;

public class BridgeTextFragment implements TextFragment {
  private final ReadableMap mFragment;

  public BridgeTextFragment(ReadableMap fragment) {
    this.mFragment = fragment;
  }

  @Override
  public TextAttributeProps getTextAttributeProps() {
    return TextAttributeProps.fromReadableMap(new ReactStylesDiffMap(mFragment.getMap("textAttributes")));
  }

  @Override
  public String getString() {
    return mFragment.getString("string");
  }

  @Override
  public boolean hasReactTag() {
    return mFragment.hasKey("reactTag");
  }

  @Override
  public int getReactTag() {
    return mFragment.getInt("reactTag");
  }

  @Override
  public boolean hasIsAttachment() {
    return mFragment.hasKey(ViewProps.IS_ATTACHMENT);
  }

  @Override
  public boolean isAttachment() {
    return mFragment.getBoolean(ViewProps.IS_ATTACHMENT);
  }

  @Override
  public double getWidth() {
    return mFragment.getDouble(ViewProps.WIDTH);
  }

  @Override
  public double getHeight() {
    return mFragment.getDouble(ViewProps.HEIGHT);
  }
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/fragments/BridgeTextFragment.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.views.text.fragments

import com.facebook.react.bridge.ReadableMap
import com.facebook.react.uimanager.ReactStylesDiffMap
import com.facebook.react.uimanager.ViewProps
import com.facebook.react.views.text.TextAttributeProps

/** A [TextFragment] implementation backed by a a [ReadableMap] */
internal class BridgeTextFragment(private val fragment: ReadableMap) : TextFragment {
  override val textAttributeProps: TextAttributeProps
    get() =
        TextAttributeProps.fromReadableMap(ReactStylesDiffMap(fragment.getMap("textAttributes")))

  override val string: String?
    get() = fragment.getString("string")

  override fun hasReactTag(): Boolean = fragment.hasKey("reactTag")

  override val reactTag: Int
    get() = fragment.getInt("reactTag")

  override fun hasIsAttachment(): Boolean = fragment.hasKey(ViewProps.IS_ATTACHMENT)

  override val isAttachment: Boolean
    get() = fragment.getBoolean(ViewProps.IS_ATTACHMENT)

  override val width: Double
    get() = fragment.getDouble(ViewProps.WIDTH)

  override val height: Double
    get() = fragment.getDouble(ViewProps.HEIGHT)
}
